stan/math/rev/fun/eigenvalues_sym.hpp

```diff
@@ -0,0 +1,117 @@
+#ifndef STAN_MATH_REV_FUN_EIGENVALUES_HPP
+#define STAN_MATH_REV_FUN_EIGENVALUES_HPP
+
+#include <stan/math/rev/meta.hpp>
+#include <stan/math/rev/core.hpp>
+#include <stan/math/rev/fun/value_of_rec.hpp>
+#include <stan/math/rev/fun/typedefs.hpp>
+#include <stan/math/prim/fun/Eigen.hpp>
+#include <stan/math/prim/err/check_symmetric.hpp>
+#include <stan/math/prim/err/check_nonzero_size.hpp>
+#include <stan/math/prim/fun/typedefs.hpp>
+#include <stan/math/prim/fun/value_of_rec.hpp>
+
+namespace stan {
+namespace math {
+
+namespace internal {
+class eigenvalues_vari : public vari {
+ public:
+  int M_;  // A.rows() = A.cols()
+  double *A_;
+  double *w_;  // eigenvalues
+  double *v_;  // eigenvectors
+  vari **vari_ref_A_;
+  vari **vari_ref_w_;
+  vari **vari_ref_v_;
+
+  explicit eigenvalues_vari(const Eigen::Matrix<var, -1, -1> &A)
+      : vari(0.0),
+        M_(A.rows()),
+        A_(reinterpret_cast<double *>(
```

Comment:
Can rewrite these as `ChainableStack::instance_->memalloc_.alloc_array<double>(A.rows() * A.cols())` so no reinterpret_cast is necessary.

Reply:
Done.